lib/sedna/core/base.py

```diff
@@ -12,21 +12,114 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-import os.path
+import os
+import threading
+import gc
+import time
 import json
 
 from sedna.common.log import LOGGER
 from sedna.common.file_ops import FileOps
 from sedna.common.config import BaseConfig
 from sedna.common.config import Context
 from sedna.common.constant import K8sResourceKind
+from sedna.common.constant import K8sResourceKindStatus
 from sedna.service.client import LCClient
 from sedna.backend import set_backend
 from sedna.common.class_factory import ClassFactory, ClassType
 
 __all__ = ('JobBase',)
 
 
+class ModelLoadingThread(threading.Thread):
+    """ MulThread support hot model loading"""
+    MODEL_MANIPULATION_SEM = threading.Semaphore(1)
+    MODEL_CHECK_TIME = max(
+        int(Context.get_parameters("MODEL_POLL_PERIOD_SECONDS", "60")), 1
+    )
+    MODEL_HOT_UPDATE_CONF = Context.get_parameters("MODEL_HOT_UPDATE_CONFIG")
+
+    def __init__(self,
+                 estimator,
+                 job_kind,
+                 job_name,
+                 worker_name,
+                 namespace,
+                 lc_server=None,
+                 version="latest"
+                 ):
+        self.production_estimator = estimator
+        self.job_name = job_name
+        self.job_kind = job_kind
+        self.namespace = namespace
+        self.worker_name = worker_name
+        self.lc_server = lc_server
+        self.version = version
+        temp_path = self.production_estimator.model_save_path or "/tmp"
+        self.temp_path = os.path.dirname(temp_path)
+        super(ModelLoadingThread, self).__init__()
+
+    def report_task_info(self, status, kind="deploy"):
```

Comment:
why is it not a common method? or use a common method? It's not necessary!

Reply:
Yep, It's a common method. you can review the latest version